ui/src/app/views/account-creation-modals/component.tsx

```diff
@@ -0,0 +1,206 @@
+import {Component, DoCheck, Input, OnInit} from '@angular/core';
+
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+import {environment} from 'environments/environment';
+
+import {
+  FetchArgs,
+  ProfileApiFetchParamCreator,
+  ResendWelcomeEmailRequest,
+  UpdateContactEmailRequest
+} from 'generated/fetch/api';
+
+import {
+  Error,
+  FieldInput,
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalTitle,
+  styles,
+} from 'app/react-components/modals';
+
+function isBlank(s: string) {
+  return (!s || /^\s*$/.test(s));
+}
+
+class AccountCreationModalsReact extends React.Component<any, any> {
+
+  state: {
+    changingEmail: boolean,
+    resendingEmail: boolean,
+    contactEmail: string,
+    emailOffFocus: boolean,
+    };
+  props: {
+    username: string,
+    creationNonce: string,
+    passNewEmail: Function,
+    update: boolean,
+    resend: boolean,
+    closeFunction: Function,
+    };
+
+  constructor(props: Object) {
+    super(props);
+    this.state = {
+      changingEmail: false,
+      resendingEmail: false,
+      contactEmail: '',
+      emailOffFocus: true,
+    };
+  }
+
+  fullUrl(url: string): string {
+    return environment.allOfUsApiUrl + url;
+  }
+
+  updateAndSend(): void {
+    if (this.contactEmailInvalidError()) {
+      return;
+    }
+    this.props.passNewEmail(this.state.contactEmail);
+    const request: UpdateContactEmailRequest = {
+      username: this.props.username,
+      contactEmail: this.state.contactEmail,
+      creationNonce: this.props.creationNonce
+    };
+    const args: FetchArgs = ProfileApiFetchParamCreator().updateContactEmail(request);
+    fetch(this.fullUrl(args.url), args.options).then(() => {
+      this.setState({resendingEmail: false, changingEmail: false});
+    });
+    this.props.closeFunction();
+  }
+
+  send() {
+    const request: ResendWelcomeEmailRequest = {
+      username: this.props.username,
+      creationNonce: this.props.creationNonce
+    };
+    const args: FetchArgs = ProfileApiFetchParamCreator().resendWelcomeEmail(request);
+    fetch(this.fullUrl(args.url), args.options).then(() => {
+      this.setState({resending: false});
+    });
+    this.props.closeFunction();
+  }
+
+  leaveFocusEmail(evt): void {
+    this.setState({emailOffFocus: true, contactEmail: evt.target.value});
```

Comment:
Rather than setting `contactEmail` here, it's more typical to use an `onChange` handler on the input, which will fire changes on every keystroke. You can still track the focus state for the purpose of toggling the error message.